### PR TITLE
Nil unbonding time

### PR DESCRIPTION
**Pull request by joe-bowman**
## 1. Summary
Fixes # QCK-500

Permit nil completion times for unbondings and redelegations.

On receipt of nil redelegation completion time, remove the redelegation record.

Additional clean up of Acknowledgement and txResponse unmarshalling.

## 2.Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] State breaking change

## 3. Implementation details

In HandleBeginUnbonding and HandleBeginRedelegation (ibc_packet_handlers.go) we remove the check for a nil completion time.

In HandleBeginRedelegation, on receipt of nil redelegation completion time, remove the redelegation record.

Use built-in ibc-go methods to inspect failure/sucess of incoming acknowledgement.

Fixes relating to delegation updates on redelegation completion.

### CODE REVIEW

**Comment by joe-bowman** [comment]
Need to lint and add test case.

File: x/interchainstaking/keeper/ibc_packet_handlers.go [hunk 1]
```diff
@@ -139,22 +130,15 @@ func (k *Keeper) HandleAcknowledgement(ctx sdk.Context, packet channeltypes.Pack
 			}
 			response := lsmstakingtypes.MsgRedeemTokensforSharesResponse{}
 
-			if msgResponseType != "" {
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/ibc_packet_handlers.go:142]: Each one of these cases was doing the same logic for msgResponse and then msgData.Data - refactored for less code and greater simplicity.

**Comment by joe-bowman** [comment]
@ajansari95 @aljo242 @muku314115 - ready for review now!

File: x/interchainstaking/keeper/ibc_packet_handlers.go [hunk 2]
```diff
@@ -109,26 +103,21 @@ func (k *Keeper) HandleAcknowledgement(ctx sdk.Context, packet channeltypes.Pack
...
 		//nolint:staticcheck // SA1019 ignore this!
-		var msgData *sdk.MsgData
 		var msgResponse []byte
-		var msgResponseType string
 		if len(txMsgData.MsgResponses) > 0 {
-			msgResponseType = txMsgData.MsgResponses[msgIndex].GetTypeUrl()
 			msgResponse = txMsgData.MsgResponses[msgIndex].GetValue()
 		} else if len(txMsgData.Data) > 0 {
-			msgData = txMsgData.Data[msgIndex]
+			msgResponse = txMsgData.Data[msgIndex].GetData()
```
Comment (aljo242) [inline x/interchainstaking/keeper/ibc_packet_handlers.go:110]: Could potentially lead to an out of bounds `panic()`?

File: x/interchainstaking/keeper/ibc_packet_handlers.go [hunk 3]
```diff
@@ -139,22 +128,15 @@ func (k *Keeper) HandleAcknowledgement(ctx sdk.Context, packet channeltypes.Pack
...
-				}
-			} else {
-				err := proto.Unmarshal(msgData.Data, &response)
-				if err != nil {
-					k.Logger(ctx).Error("unable to unmarshal MsgRedeemTokensforShares response", "error", err)
-					return err
-				}
+			err = proto.Unmarshal(msgResponse, &response)
+			if err != nil {
+				k.Logger(ctx).Error("unable to unpack MsgRedeemTokensforShares response", "error", err)
```
Comment (aljo242) [inline x/interchainstaking/keeper/ibc_packet_handlers.go:133]:
```suggestion
				k.Logger(ctx).Error("unable to unmarshal MsgRedeemTokensforShares response", "error", err)
```

consistency

**Review by ajansari95 — APPROVED** [review]
lgtm

**Review by ajansari95 — COMMENTED** [review]
lint failing

**Review by aljo242 — APPROVED** [review]
lgtm

**Review by ajansari95 — APPROVED** [review]
lgtm

**Pull request merged**